### DISCUSSION TITLE
Setting default config values when using pocketbase in framework mode

### DISCUSSION THREAD

**Discussion by M-Porter** [pocketbase/pocketbase#587] [Q&A] [answered]
Hi, messing around with pocketbase for the first time to see if it's a good fit for my application.

There's some default values (e.g. app name, url, disable collection create UI, etc etc) that i'd like to be able to set on application boot via env values.

This is what i'm trying to do:
```go
package main

import (
	"log"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	app.Settings().Merge(&core.Settings{
		Meta: core.MetaConfig{
			AppName: "My App Name",
		},
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

After booting the server and checking the application settings, the app name is still `Acme`.

Also to rule out dirty data, I deleted the `pb_data` directory and the name still was not the expected `My App Name`.

What is the best way to set default values such as this for framework flavored applications?

**Comment by ganigeorgiev** [comment, marked as answer]
This could be done via DB migration. Currently the documentation is lacking on this, but you could check this guide - https://pocketbase.io/docs/db-migrations/. The settings are stored as json in the `_params` table, but the problem is that they can be encrypted (when the `--encryptionEnv` variable is set) so it may not be possible to just update the record.

I'll consider in a future release to add a helper hook that will be called after the first settings load (aka. after decrypt and load into the app instance).

Alternatively, you can also attach to the `OnBeforeServe` hook and modify the settings in place. Something like this (_I haven't tested it_):

```go
package main

import (
	"log"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		// note: This will overwrite any previously loaded app settings value in place.
		//       You may want to add some checks when to apply it or use a db migration
		//       so that it can run only once.
		app.Settings().Merge(&core.Settings{
			Meta: core.MetaConfig{
				AppName: "My App Name",
			},
		})

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

**Reply by M-Porter** [reply]
Awesome. Thanks for the recommendation.
